lib/src/merged_tree.rs

```diff
@@ -317,7 +317,8 @@ impl MergedTree {
 
 /// A single entry in a tree diff.
 pub struct TreeDiffEntry {
-    // pub source: RepoPathBuf,
+    /// The source path.
+    pub source: RepoPathBuf,
```

Comment:
I wonder if it can be measurable slower to have this field here when it's typically the same as `target`. Maybe we should make this `Option<RepoPathBuf>` and not set it if it's the same as `target`. Just a thought. Let's leave it as is until we've noticed that it makes a difference.

Reply:
while that was the original design, it makes interacting with the thing more cumbersome, so I went with the simplest thing I could for this rework.